src/apps/tempus/helpers/tempus.pool.token-helper.ts

```diff
@@ -0,0 +1,46 @@
+import { Inject, Injectable } from '@nestjs/common';
+import _ from 'lodash';
+
+import { IAppToolkit, APP_TOOLKIT } from '~app-toolkit/app-toolkit.interface';
+import { Network } from '~types/network.interface';
+
+import { TempusContractFactory, PyToken } from '../contracts';
+import { TEMPUS_DEFINITION } from '../tempus.definition';
+
+import { getTempusData, TempusPool } from './tempus.datasource';
+
+const appId = TEMPUS_DEFINITION.id;
+const groupId = TEMPUS_DEFINITION.groups.pool.id;
+
+@Injectable()
+export class TempusTokensTokenFetcher {
+  constructor(
+    @Inject(APP_TOOLKIT) private readonly appToolkit: IAppToolkit,
+    @Inject(TempusContractFactory) private readonly contractFactory: TempusContractFactory,
+  ) {}
+
+  async getPoolTokens(data: TempusPool, network: Network) {
+    return await this.appToolkit.helpers.vaultTokenHelper.getTokens<PyToken>({
+      appId,
+      groupId,
+      network,
+      resolveVaultAddresses: () => [data.principalsAddress.toLowerCase(), data.yieldsAddress.toLowerCase()],
+      resolveContract: ({ address, network }) => this.contractFactory.pyToken({ address, network }),
+      resolveUnderlyingTokenAddress: () => data.backingTokenAddress.toLowerCase(),
+      resolveReserve: () => 0,
+      resolvePricePerShare: async ({ multicall, contract }) => {
+        const decimals = await multicall.wrap(contract).decimals();
+        const pricePerShare = await multicall.wrap(contract).getPricePerFullShareStored();
+        return Number(pricePerShare) / 10 ** Number(decimals);
+      },
```

Comment:
I think the default label isn't great: `TYS-stETH-31-2-2022`

I'd suggest we override the label with `resolvePrimaryLabel`, and set it to `stETH Yield Share (Maturity: 31/02/2022)` (or for principal, `stETH Principal Share (Maturity: 31/02/2022)`). You can also adjust the vault helper to add a `resolveTertiaryLabel` if you want the maturity outside of the primary label.

Reply:
Yup it's ok to come back and address this later. I'll update the `VaultTokenHelper` between now and then so that you'll be able to update the `tertiaryLabel` in the future if necessary.